### PR TITLE
Find important features

### PR DESCRIPTION
**Pull request by marco-rossi29**
Usage example:
`>python vw_important_features.py -d data.json --ml_args "--cb_adf -l 0.01 --cb_type mtr"`

Sample output:
```
=====================================
Testing a range of L1 regularization
L1: 1e-06 - Num of Features: 4197
L1: 1e-05 - Num of Features: 2741
L1: 1e-04 - Num of Features: 46
L1: 1e-03 - Num of Features: 13
=====================================

Inverting hashes of important features (l1 = 1e-03)
=====================================
Emotion0^contempt:194265:-0.0236295
Emotion0^disgust:251874:-0.0327635
Emotion0^fear:188970:1.0217
Emotion0^surprise:1982:-0.416561
Emotion1^anger:197213:-0.486541
Emotion1^contempt:94653:-0.34001
Emotion1^disgust:166742:-0.368214
Emotion1^fear:23093:-5.25565
Emotion2^anger:104155:-0.20074
Emotion2^contempt:104000:-0.457823
Emotion2^disgust:22454:-0.142982
Emotion2^fear:23283:-177.874
Emotion2^sadness:157471:-1.35735
=====================================
```

### CODE REVIEW

**Comment by JohnLangford** [comment]
These are good suggestions, thanks @arielf 

It might also be nice to have a test that invokes this in RunTests as it seems like a common use case.

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging 687c1556254e6fe0d31f44132be455860210e434 into eb63f84a6c99e9e98a221189912babd236f14d89 - [view on LGTM.com](https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-2c9bf008bc28d7f59193887e2169b53ac6f47e88)

**new alerts:**

* 1 for &#39;input&#39; function used

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by marco-rossi29** [comment]
@arielf Hi Ariel,
How are you?
Thank you for your comments. I've updated a new version fixing things.

Let me know if you have other improvements!
Marco

**Comment by JohnLangford** [comment]
lgtm is complaining about input: https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-2c9bf008bc28d7f59193887e2169b53ac6f47e88

And, is a test case viable?  Something like 152: https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/RunTests#L1554 except with this script instead?

**Comment by marco-rossi29** [comment]
> lgtm is complaining about input: https://lgtm.com/projects/g/VowpalWabbit/vowpal_wabbit/rev/pr-2c9bf008bc28d7f59193887e2169b53ac6f47e88

This is a script for testing, I think having `input` is necessary for clarity

> 
> And, is a test case viable? Something like 152: https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/RunTests#L1554 except with this script instead?

What kind of test do you have in mind? Is it to unit-test this script, or to test the system itself?

**Comment by JohnLangford** [comment]
The input issue sounds like a good thing to avoid: https://lgtm.com/rules/3980089/   Can we use raw_input() instead?

I was imagining something more like an end-to-end test of the script (which also documents its use) using one of the small existing datasets in the test directory.

**Comment by marco-rossi29** [comment]
> The input issue sounds like a good thing to avoid: https://lgtm.com/rules/3980089/ Can we use raw_input() instead?
raw_input() was renamed input() in python3

> I was imagining something more like an end-to-end test of the script (which also documents its use) using one of the small existing datasets in the test directory.
Sure, I'll work on that.

**Comment by JohnLangford** [comment]
@marco-rossi29 , this is primarily waiting on a test example (which should be easy).  

As far as input(), I'm not sure what to do.  Does python3 still have a raw_input()?  If so, maybe that's safer since it still does the right thing in earlier python?  (It would be kind of nice to avoid growing the number of LGTM issues regardless of whether or not they are legitimate.)

File: utl/vw-important-features [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# ==========================================================================================
+# Find important features
+# ==========================================================================================
+
+'''
+Usage example:
+>python vw-important-features -d data.json --ml_args "--cb_adf -l 0.01 --cb_type mtr"
```
Comment (arielf) [inline utl/vw-important-features]: If the file is made executable (`chmod +x vw-important-features`) before check-in, there's no need to prefix the call with `python`

File: utl/vw-important-features [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Testing a range of L1 regularization
+L1: 1e-06 - Num of Features: 4197
+L1: 1e-05 - Num of Features: 2741
+L1: 1e-04 - Num of Features: 46
+L1: 1e-03 - Num of Features: 13
+=====================================
+
+Inverting hashes of important features (l1 = 1e-03)
+=====================================
+Emotion0^contempt:194265:-0.0236295
```
Comment (arielf) [inline utl/vw-important-features]: Wouldn't it make more sense to sort the output descending by `abs(weight)` ?

Comment (marco-rossi29) [reply utl/vw-important-features]: That's a good suggestion. Right now I'm not even parsing the values. But it can be done in the future.

**Comment by marco-rossi29** [comment]
Closing this because duplicate file is on https://github.com/Microsoft/mwt-ds/pull/132

**Pull request closed without merging**